realm/src/main/java/io/realm/Realm.java

```diff
@@ -1115,11 +1112,23 @@ boolean contains(Class<?> clazz) {
      */
     public <E extends RealmObject> RealmResults<E> allObjectsSorted(Class<E> clazz, String fieldNames[],
                                                                boolean sortAscending[]) {
-        // FIXME: This is not an optimal implementation. When core's Table::get_sorted_view() supports
-        // FIXME: multi-column sorting, we can rewrite this method to a far better implementation.
-        RealmResults<E> results = this.allObjects(clazz);
-        results.sort(fieldNames, sortAscending);
-        return results;
+        validateMultiSortParameters(fieldNames, sortAscending);
```

Comment:
At C++ we only check that the lengths of the two arrays are equal but we can easily move all checks to C++.